### ISSUE TITLE
Scrum Updates unit5

### ISSUE THREAD

**Issue by Ivyqiuhan** [mattbekh/frens-app#35]
Due 10pm Jul 18th

1. What did you work on this past iteration (2 weeks)?
2. What were any major issues/challenges you ran into?
3. What do you plan to work on for this coming iteration (2 weeks)?

Group members:

- Danny Deng k6s2b
- Ivy He y6w2b
- Matt Bekhterev t8h7w
- Michaux Sun m5j2b

**Comment by Ivyqiuhan** [comment]
1. This week I have been working on database related stuff, such as creating the database schema/actual data and connect database to the backend.
2. Creating the database is relative straight forward, but updating the node_modules between group members are very tricky, since we both have Windows and Mac in the group, we need to update/remove node_modules every time we pull from the project branch, very frustrating. We need to find a smarter way to manage it.
3. For the next iteration, we are going to focus more on our stretch goals: chat functionality and matching algorithm. I will do some research on both topics, but I want to focus more on implementing machine learning cluster methods

**Comment by mattbekh** [comment]
Matt Bekhterev (t8h7w)

1. This week I created the backend architecture, hooked up to MongoDB using Atlas, routed endpoints using axios, installed diagnostic library Morgan as well as create a custom error class to better identify errors and server calls. I also worked on the user backend models and schemas, and assisted with password encryption using the Bcrypt library.
2. Due to some asynchronous behavior, redirecting the user after registration/ login sometimes caused errors because the data would not flow back in time for render. This was a tricky problem to solve which I ended up solving with a history push to direct a user to the main page in a callback.
3. The goal for the next iteration is to make the project file presentable, clean up the architecture of the code and focus on the stretch goals of matching people with others of similar interests and to create the chat functionality.

**Comment by dannyd2083** [comment]
Danny Deng (k6s2b)

1.This week I created the validation of the input for the register page so that user can't just submit empty form to the database. I also change the Interest Box to Drop down selection.
2. It took me times to learn how to properly styling React Select with Styled-components. The dropdown menu was blocked by the Middle UI area before.
3. The left missions for the project are matching and chat functionality (extra), I will focus on how to match users properly.

**Comment by MichauxS-cc** [comment]
1. This sprint, I worked on using json web token to keep track of the logged in user between different web pages. Combining with local storage, I was able to display the corresponding user info on Main page once the user has register/login.
2. Figuring out how to store and retrieve the token took me a while since it requires a specific format. I ended up creating a customized header obj in request from client side and pass it to the backend. Also, authenticate the token regarding user's status (user from register/login page) is tricky which I solved by checking the user status using object chaining.
3. For the next iteration, our goal is to add the function to allow user to logout, create the matching algorithm to match people more precise, and achieve the stretch goal of chatting functionality.
